jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/ReopenCaseCommand.java

```diff
@@ -78,29 +78,43 @@ public Void execute(Context context) {
         
         KieSession ksession = ((RegistryContext) context).lookup( KieSession.class );
                                
-        CaseFileInstance caseFile = getCaseFile(ksession, caseId);                        
+        CaseFileInstance caseFile = getCaseFile(ksession, caseId);     
         
         caseEventSupport.fireBeforeCaseReopened(caseId, caseFile, deploymentId, caseDefinitionId, data);
         
         logger.debug("Updating case file in working memory");
         FactHandle factHandle = ksession.getFactHandle(caseFile);
-        ((CaseFileInstanceImpl)caseFile).setCaseReopenDate(new Date());
-        if (data != null && !data.isEmpty()) {
-            caseFile.addAll(data);
-        }
         ksession.update(factHandle, caseFile);
         
         logger.debug("Starting process instance for case {} and case definition {}", caseId, caseDefinitionId);
         CorrelationKey correlationKey = correlationKeyFactory.newCorrelationKey(caseId);
         Map<String, Object> params = new HashMap<>();
         // set case id to allow it to use CaseContext when creating runtime engine
         params.put(EnvironmentName.CASE_ID, caseId);
+        final Map<String, Object> caseData = caseFile.getData();
+        
+        
+        for (Map.Entry<String, Object> entry : caseData.entrySet()) {
+            params.put(VariableScope.CASE_FILE_PREFIX + entry.getKey(), entry.getValue());
+        }
+        
+        if (data != null)   {
+            for (Map.Entry<String, Object> entry : data.entrySet()) {
+                params.put(VariableScope.CASE_FILE_PREFIX + entry.getKey(), entry.getValue());
+            }
+        }
+
         long processInstanceId = processService.startProcess(deploymentId, caseDefinitionId, correlationKey, params);
+        
+        ((CaseFileInstanceImpl)caseFile).setCaseReopenDate(new Date());
+        if (data != null) {
+            caseFile.addAll(data);
+        }
+        
         logger.debug("Removing case file from working memory to allow refiring of rules...");
         ksession.delete(factHandle);
         ksession.insert(caseFile);
-        final Map<String, Object> caseData = caseFile.getData();
-        if (caseData != null && !caseData.isEmpty()) {
+        if (!caseData.isEmpty()) {
```

Comment:
why removed the check of null ?

Reply:
because it was unneded and reported as a code smell